### PR TITLE
Axisymmetric mesh and targeted symmetry family updates

### PR DESCRIPTION
**Pull request by lamkina**
## Purpose

This PR adds the CGNS BC indices for axisymmetric and polar topologies. Additionally, this adds the capability to target specific mesh families when setting a symmetric plane nodes to zero.

## Expected time until merged

2-3 weeks

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: cgnsutilities/cgnsutilities.py [hunk 1]
```diff
@@ -1858,7 +1863,7 @@ def extrude(self, direction):
         """Extrudes from 2D panar grid to 3D"""
 
         # Get the data order and new dims
-        order, newDims = self._extrudeGetDataOrderAndDIms(direction)
+        order, newDims = self._extrudeGetDataOrderAndDIms(direction, 1)
```
Comment (eirikurj) [inline cgnsutilities/cgnsutilities.py:1866]: This is conflicting, and step value should be 2.

File: cgnsutilities/cgnsutilities.py [hunk 2]
```diff
@@ -2374,13 +2380,16 @@ def randomize(self, keepRHS):
...
 
-    def symmZero(self, idir):
+    def symmZero(self, idir, family=None):
         for bc in self.bocos:
-            if bc.internalType == "bcsymmetryplane":
-                # 'r' is the range. We need to subtract off -1 from
-                # the low end since it was in fortran 1-based ordering
-                r = bc.ptRange.copy()
-                self.coords[r[0, 0] - 1 : r[0, 1], r[1, 0] - 1 : r[1, 1], r[2, 0] - 1 : r[2, 1], idir] = 0.0
+            if "symmetry" in bc.internalType:
```
Comment (eirikurj) [inline cgnsutilities/cgnsutilities.py:2385]: There are only two standard CGNS bc types defined that have `symmetry` in the name. I would prefer making this explicit and do a full check and just adding the other options instead of this partial check.

**Review by eirikurj — CHANGES_REQUESTED** [review]
Did a quick skim over this. Let me know if you have questions and we can discuss.

File: cgnsutilities/cgnsutilities.py (hunk 1 above)
Comment (lamkina) [reply cgnsutilities/cgnsutilities.py]: fixed

File: cgnsutilities/cgnsutilities.py (hunk 2 above)
Comment (lamkina) [reply cgnsutilities/cgnsutilities.py]: Made the check explicit.

**Comment by lamkina** [comment]
Addressed the comments from @eirikurj . I also added two small tests that check if the z-coordinate is actually zero after running the `symmZero` method with and without a target family.

**Review by eirikurj — APPROVED** [review]
